File: client/Kronos.Service/Interface/IApplicationService.cs
using Kronos.Domain.Dto.Application;

namespace Kronos.Service.Interface
{
    public interface IApplicationService
    {
        Task<ApplicationResponseDTO?> Create(ApplicationCreateRequestDTO request);
        Task<ApplicationResponseDTO?> Update(ApplicationUpdateRequestDTO request);
        Task<bool> Delete(Guid applicationId);
        Task<List<ApplicationResponseDTO>> Get();
    }
}

File: client/lib/app/core/interface/iapplication_service.dart
import 'package:kronos/app/core/model/application/application_request.dart';
import 'package:kronos/app/core/model/application/application_response.dart';
import 'package:kronos/app/shared/tool/result.dart';

abstract class IApplicationService {
  Future<Result<List<ApplicationResponse>, String>> get();
  Future<Result<bool, String>> create(ApplicationRequest request);
  Future<Result<bool, String>> update(ApplicationRequest request);
  Future<Result<bool, String>> delete(String id);
}
